pkg/quarterdeck/tokens/validator.go

```diff
@@ -0,0 +1,61 @@
+package tokens
+
+import (
+	"fmt"
+
+	jwt "github.com/golang-jwt/jwt/v4"
+)
+
+// Validators are able to verify that access and refresh tokens were issued by
+// Quarterdeck and that their claims are valid (e.g. not expired).
+type Validator interface {
+	// Verify an access or a refresh token after parsing and return its claims.
+	Verify(tks string) (claims *Claims, err error)
+
+	// Parse an access or refresh token without verifying claims (e.g. to check an expired token).
+	Parse(tks string) (claims *Claims, err error)
```

Comment:
Does `Verify` fill out more of the claims than `Parse`? In other words will it be obvious in downstream processing if we accidentally use the wrong method (since they have the same interface).

Reply:
Unfortunately no - in 98% of use cases users want to use `Verify` -- `Parse` is a very special case and should rarely be used; but it won't be possible to tell from the claims which method was used.